cpp/lib/ome/bioformats/tiff/Util.cpp

```diff
@@ -0,0 +1,446 @@
+          }
+        core.sizeZ = core.sizeT = core.imageCount = 1U;
+
+        // libtiff does any needed endian conversion
+        // automatically, so the data is always in the native
+        // byte order.
+#ifdef BOOST_BIG_ENDIAN
+        core.littleEndian = false;
+#else // ! BOOST_BIG_ENDIAN
+        core.littleEndian = true;
+#endif // BOOST_BIG_ENDIAN
+
+        // This doesn't match the reality, but since subchannels are
+        // addressed as planes this is needed.
+        core.interleaved = false;
+
+        // Indexed samples.
+        if (samples == 1 && photometric == tiff::PALETTE)
+          {
+            try
+              {
+                ome::compat::array<std::vector<uint16_t>, 3> cmap;
+                ifd.getField(tiff::COLORMAP).get(cmap);
+                core.indexed = true;
+                core.rgb = false;
+              }
+            catch (...)
+              {
+              }
+          }
+        // Indexed samples for different photometric interpretations;
+        // not currently supported fully.
+        else
+          {
+            try
+              {
+                uint16_t indexed;
+                ifd.getField(tiff::INDEXED).get(indexed);
+                if (indexed)
+                  {
+                    core.indexed = true;
+                    core.rgb = false;
+                  }
+              }
+            catch (...)
+              {
+              }
+          }
+
+        // Add series metadata from tags.
+        setMetadata(ifd, core, "PageName #", PAGENAME);
+        setMetadata(ifd, core, "ImageWidth", IMAGEWIDTH);
+        setMetadata(ifd, core, "ImageLength", IMAGELENGTH);
+        setMetadata(ifd, core, "BitsPerSample", BITSPERSAMPLE);
+
+        /// @todo EXIF IFDs
+
+        setMetadata(ifd, core, "PhotometricInterpretation", PHOTOMETRIC);
+
+        /// @todo Text stream output for Tag enums.
+        /// @todo Metadata type for PhotometricInterpretation.
+
+        try
+          {
+            setMetadata(ifd, core, "Artist", ARTIST);
+            Threshholding th;
+            ifd.getField(THRESHHOLDING).get(th);
+            core.seriesMetadata.set("Threshholding", th);
+            if (th == HALFTONE)
+              {
+                setMetadata(ifd, core, "CellWidth", CELLWIDTH);
+                setMetadata(ifd, core, "CellLength", CELLLENGTH);
+              }
+          }
+        catch (...)
+          {
+          }
+
+        setMetadata(ifd, core, "Orientation", ORIENTATION);
+
+        /// @todo Image orientation (storage order and direction) from
+        /// ORIENTATION; fix up width and length from orientation.
+
+        setMetadata(ifd, core, "SamplesPerPixel", SAMPLESPERPIXEL);
+        setMetadata(ifd, core, "Software", SOFTWARE);
+        setMetadata(ifd, core, "Instrument Make", MAKE);
+        setMetadata(ifd, core, "Instrument Model", MODEL);
+        setMetadata(ifd, core, "Make", MAKE);
+        setMetadata(ifd, core, "Model", MODEL);
```

Comment:
It's a direct copy of the `BaseTiffReader.java` behaviour; it's unchanged in this PR (just moved from IFD.cpp)